src/api/useFetch.js

```diff
@@ -98,9 +101,12 @@ const _useFetch = ({
         const cachedResult = cache.get(cacheKey)
         if (cachedResult) {
           try {
-            const result = cachedResult instanceof Promise
-              ? await cachedResult
-              : cachedResult
+            let result = cachedResult
+
+            if (cachedResult instanceof Promise) {
+              if (!isConnected) return // assume promise can't be resolved without connection
```

Comment:
I think `isFetching.current` has to be set to `false` again in case fetching is skipped here, otherwise we won't be able to fetch the result again in case the connection is restored.